graphql/ui_plugin_build_baron.go

```diff
@@ -269,23 +268,20 @@ func BbGetProject(settings *evergreen.Settings, projectId string) (evergreen.Bui
 func BbGetTask(taskId string, executionString string) (*task.Task, error) {
 	execution, err := strconv.Atoi(executionString)
 	if err != nil {
-		return nil, errors.Wrap(err, "Invalid execution number")
+		return nil, errors.Wrap(err, "invalid execution number")
 	}
+
 	t, err := task.FindOneIdOldOrNew(taskId, execution)
 	if err != nil {
-		return nil, errors.Wrap(err, "problem finding task")
+		return nil, errors.Wrap(err, "finding task")
 	}
 	if t == nil {
-		return nil, errors.Errorf("No task found for taskId: %s and execution: %d", taskId, execution)
+		return nil, errors.Errorf("no task found for task id: %s and execution: %d", taskId, execution)
 	}
-	if t.DisplayOnly {
-		t.LocalTestResults, err = t.GetTestResultsForDisplayTask()
-		if err != nil {
-			return nil, errors.Wrapf(err, "Problem finding test results for display task '%s'", t.Id)
-		}
-	}
-	return t, nil
+
+	return t, errors.Wrap(t.PopulateTestResults(), "populating test results")
```

Comment:
We should split this up. We typically don't return a value along side errors. We prefer to return nil values when theres an error since we propagate that error to the UI. 
```suggestion
        err = t.PopulateTestResults()
        if err != nil {
             return nil, errors.Wrap(err, "Error while populating test results")
         }
	return t, nil
```